shell/common/gin_converters/usb_protected_classes_converter.h

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2022 Microsoft, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#ifndef ELECTRON_SHELL_COMMON_GIN_CONVERTERS_USB_PROTECTED_CLASSES_CONVERTER_H_
+#define ELECTRON_SHELL_COMMON_GIN_CONVERTERS_USB_PROTECTED_CLASSES_CONVERTER_H_
+
+#include <string>
+#include <string_view>
+#include <utility>
+#include <vector>
+
+#include "gin/converter.h"
+#include "services/device/public/mojom/usb_device.mojom-forward.h"
+#include "shell/browser/electron_permission_manager.h"
+
+namespace gin {
+
+static auto constexpr ClassMapping =
+    std::array<std::pair<uint8_t, std::string_view>, 7>{
+        {{device::mojom::kUsbAudioClass, "audio"},
+         {device::mojom::kUsbHidClass, "hid"},
+         {device::mojom::kUsbMassStorageClass, "mass_storage"},
+         {device::mojom::kUsbSmartCardClass, "smart_card"},
+         {device::mojom::kUsbVideoClass, "video"},
+         {device::mojom::kUsbAudioVideoClass, "audio_video"},
+         {device::mojom::kUsbWirelessClass, "wireless"}}};
+
+template <>
+struct Converter<electron::ElectronPermissionManager::USBProtectedClasses> {
+  static v8::Local<v8::Value> ToV8(
+      v8::Isolate* isolate,
+      const electron::ElectronPermissionManager::USBProtectedClasses& classes) {
+    std::vector<std::string> class_strings;
+    for (auto itr : classes) {
+      for (auto const& [usb_class, name] : ClassMapping) {
+        if (usb_class == itr)
+          class_strings.push_back(static_cast<std::string>(name));
```

Comment:
_(nits)_

- can `const&` the `classes` iteration
- can use `emplace_back` to avoid the `std::string` temporary
- [west const](https://hackingcpp.com/cpp/design/east_vs_west_const.html) for consistency with Chromium, I think I caused this wart via my code snippet :slightly_smiling_face: 
 
```suggestion
    std::vector<std::string> class_strings;
    class_strings.reserve(std::size(classes));
    for (const auto& itr : classes) {
      for (const auto& [usb_class, name] : ClassMapping) {
        if (usb_class == itr)
          class_strings.emplace_back(name);
```